diff --git a/tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp b/tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp
index 79e392c901..ea29e3fa4e 100644
--- a/tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp
+++ b/tracking/include/pcl/tracking/impl/kld_adaptive_particle_filter.hpp
@@ -82,7 +82,7 @@ pcl::tracking::KLDAdaptiveParticleFilterTracker<PointInT, StateT>::resample ()
       ++k;
     ++n;
   }
-  while (k < 2 || (n < maximum_particle_number_ && n < calcKLBound (k)));
+  while (n < maximum_particle_number_ && (k < 2 || n < calcKLBound (k)));
   
   particles_ = S;               // swap
   particle_num_ = static_cast<int> (particles_->points.size ());
